src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -435,10 +486,16 @@ class ThetaStarPathPlanner : public PathPlanner
     // isUnblocked and lineOfSight.
     // Description of the Grid-
     // unblocked_grid is indexed with coordinate
-    // true --> The cell is not blocked
+    // true  --> The cell is not blocked
     // false --> The cell is blocked
     // We update this as we go to avoid updating cells we don't use
-    std::map<Coordinate, bool> unblocked_grid;
+    std::unordered_map<Coordinate, bool, CoordinateHashFunction> unblocked_grid;
```

Comment:
we can get rid of this and also isUnblocked apparently isn't being used

Reply:
isUnblocked is actually still being used in `adjustEndPointsAndCheckForNoPath` and `findClosestUnblockedCell` since `rasterize` is called after the initial checks of `findPath`, so we still rely on isUnblocked slightly. I will remove the unblocked_grid as it shouldn't have any significant effect on performance anymore.